models/inception_v4/model.py

```diff
@@ -0,0 +1,385 @@
+
+        x2_0 = self.branch2_0(x)
+        x2_1 = self.branch2_1(x2_0)
+        x2_2 = self.branch2_2(x2_1)
+        x2_3a = self.branch2_3a(x2_2)
+        x2_3b = self.branch2_3b(x2_2)
+        x2 = torch.cat((x2_3a, x2_3b), 1)
+
+        x3 = self.branch3(x)
+
+        out = torch.cat((x0, x1, x2, x3), 1)
+        return out
+
+
+class Inception(nn.Module):
+    """Inception v4 model architecture from
+    `"Inception-v4, Inception-ResNet and the Impact of Residual Connections on Learning" <https://arxiv.org/abs/1602.07261>`_.
+    .. note::
+        **Important**: In contrast to the other models the inception_v4 expects tensors with a size of
+        N x 3 x 299 x 299, so ensure your images are sized accordingly.
+    """
+    def __init__(self, num_classes=1001):
+        super(Inception, self).__init__()
+        # Special attributs
+        self.input_space = None
+        self.input_size = (299, 299, 3)
+        self.mean = None
+        self.std = None
+        # Modules
+        self.features = nn.Sequential(
+            BasicConv2d(3, 32, kernel_size=3, stride=2),
+            BasicConv2d(32, 32, kernel_size=3, stride=1),
+            BasicConv2d(32, 64, kernel_size=3, stride=1, padding=1),
+            Mixed_3a(),
+            Mixed_4a(),
+            Mixed_5a(),
+            Inception_A(),
+            Inception_A(),
+            Inception_A(),
+            Inception_A(),
+            Reduction_A(), # Mixed_6a
+            Inception_B(),
+            Inception_B(),
+            Inception_B(),
+            Inception_B(),
+            Inception_B(),
+            Inception_B(),
+            Inception_B(),
+            Reduction_B(), # Mixed_7a
+            Inception_C(),
+            Inception_C(),
+            Inception_C()
+        )
+        self.last_linear = nn.Linear(1536, num_classes)
+
+    def logits(self, features):
+        #Allows image of any size to be processed
+        adaptiveAvgPoolWidth = features.shape[2]
+        x = F.avg_pool2d(features, kernel_size=adaptiveAvgPoolWidth)
+        x = x.view(x.size(0), -1)
+        x = self.last_linear(x)
+        return x
+
+    def forward(self, input):
+        x = self.features(input)
+        x = self.logits(x)
+        return x
+
+
+class InceptionV4:
+    """Initialize Inceptionv4 model.
+    """
+    def __init__(self, num_classes=1000, pretrained='imagenet') -> None:
+        self._model = None
+        self.load_model(num_classes, pretrained)
+        
+        # Set model to eval mode
+        self._model.cuda()
+        self._model.eval()
+
+    @property
+    def model(self) -> nn.Module:
+        """Getter for the model
+
+        Returns:
+            nn.Module: torch model
+        """
+        return self._model
+
+    def print_summary(self) -> None:
+        """Print summary of the model.
+        """
+        print(summary(self._model, input_size=(3, 299, 299)))
+
+    def load_model(self, num_classes, pretrained) -> None:
```

Comment:
- Do you want users to call this method after object creation? If not, then would it be a better idea to make this method protected/private? 
- We should add type info to parameters (ex num_classes:int, etc.)